php/libraries/Candidate.class.inc

```diff
@@ -756,38 +796,30 @@ class Candidate
     static function _generateExternalID()
     {
         $factory = NDB_Factory::singleton();
+        $db      = $factory->database();
         $config  = $factory->config();
 
-        $ExternalIDSettings = $config->getSetting('ExternalID');
-        $ExternalIDArrays   = Candidate::generateArrays($ExternalIDSettings);
-
-        $db = $factory->database();
-
-        do {
-            // generate a new external ID
-            if ($ExternalIDSettings['generation'] == 'sequential') {
-                if (!empty($externalID)) {
-                    $externalID++;
-                } else {
-                    $externalID = Candidate::generateFirstStringFromArrays(
-                        $ExternalIDArrays
-                    );
-                }
-            } else {
-                $externalID = Candidate::generateRandomStringFromArrays(
-                    $ExternalIDArrays
+        return self::_generateId(
+            "External Id",
```

Comment:
That name is just a front facing thing, I believe. So, I'm okay with whatever users would rather see.